src/robotFactory.js

```diff
@@ -1,10 +1,93 @@
 'use strict';
 
-class BaseRobot {}
+function setDefaultCoords(coords, zIncluded = false) {
+  if (coords === {} || !coords) {
+    return {
+      x: 0, y: 0,
+    };
+  }
 
-class FlyingRobot {}
+  if (!coords.x) {
+    coords.x = 0;
+  }
 
-class DeliveryDrone {}
+  if (!coords.y) {
+    coords.y = 0;
+  }
+
+  if (zIncluded && !coords.z) {
+    coords.z = 0;
+  }
+
+  return coords;
+}
+
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = setDefaultCoords(coords);
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name},`
+    + ` Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords = setDefaultCoords(coords, true);
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords,
+    chipVersion, maxLoadWeight, currentLoad = null) {
+    super(name, weight, coords, chipVersion);
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if (this.currentLoad !== null || cargo.weight > this.maxLoadWeight) {
```

Comment:
![image](https://github.com/mate-academy/js_robot_factory/assets/84296041/0b24046a-768e-4016-b999-51150227ba23)

```suggestion
    if (!this.currentLoad && cargo.weight <= this.maxLoadWeight) {
```